applications/OptimizationApplication/python_scripts/algorithms/NLOPT_algorithms.py

```diff
@@ -0,0 +1,184 @@
+
+        # nlopt algorithm
+        self.algorithm_name = NLOPT_settings["algorithm_name"].GetString()
+        if(len(self.__constraints)==0):
+            self.CheckOptimizerSupport(self.algorithm_name,None)
+        else:
+            for constraint in self.__constraints:
+                if constraint.IsEqualityType():
+                    self.CheckOptimizerSupport(self.algorithm_name,"equality")
+                else:
+                    self.CheckOptimizerSupport(self.algorithm_name,"inequality")
+
+        # stopping
+        self.stopping_criteria = NLOPT_settings["stopping_criteria"]
+        self.stopping_criteria.ValidateAndAssignDefaults(self.GetDefaultParameters()["NLOPT_settings"]["stopping_criteria"])
+
+    def GetMinimumBufferSize(self) -> int:
+        return 2
+
+    def Check(self):
+        pass
+
+    def Initialize(self):
+        self.converged = False
+        self.__objective.Initialize()
+        self.__objective.Check()
+        for constraint in self.__constraints:
+            constraint.Initialize()
+            constraint.Check()
+        self.master_control.Initialize()
+        self.__control_field = self.master_control.GetControlField()
+        self.algorithm_data = ComponentDataView("algorithm", self._optimization_problem)
+
+        # create nlopt optimizer
+        self.x0 = self.__control_field.Evaluate()
+        self.nlopt_optimizer = nlopt.opt(self.GetOptimizer(self.algorithm_name), self.x0.size)
+
+        # assign objectives and constarints
+        self.nlopt_optimizer.set_min_objective(self.__objective.CalculateStandardizedValueAndGradients)
+        for constraint in self.__constraints:
+            if constraint.IsEqualityType():
+                self.nlopt_optimizer.add_equality_constraint(lambda x,grad: constraint.CalculateStandardizedValueAndGradients(x,grad),1e-8)
+            else:
+                self.nlopt_optimizer.add_inequality_constraint(lambda x,grad: constraint.CalculateStandardizedValueAndGradients(x,grad),1e-8)
+        # now set stopping criteria
+        self.nlopt_optimizer.set_ftol_rel(self.stopping_criteria["objective_rel_tol"].GetDouble())
+        self.nlopt_optimizer.set_ftol_abs(self.stopping_criteria["objective_abs_tol"].GetDouble())
+        self.nlopt_optimizer.set_xtol_rel(self.stopping_criteria["controls_rel_tol"].GetDouble())
+        self.nlopt_optimizer.set_xtol_abs(self.stopping_criteria["controls_abs_tol"].GetDouble())
+        self.nlopt_optimizer.set_maxeval(self.stopping_criteria["maximum_function_evalualtion"].GetInt())
+
+    def Finalize(self):
+        self.__objective.Finalize()
+        for constraint in self.__constraints:
+            constraint.Finalize()
+        self.master_control.Finalize()
+
+    def Solve(self):
+        self.nlopt_optimizer.optimize(self.x0)
+        CallOnAll(self._optimization_problem.GetListOfProcesses("output_processes"), Kratos.OutputProcess.PrintOutput)
```

Comment:
```suggestion
        map(Kratos.OutputProcess.PrintOutput, self._optimization_problem.GetListOfProcesses("output_processes")
```

Reply:
if you do not mind, for being consistent with the rest of the application I would keep it

Reply:
I dont mind, but oh boy, reimplementing language features :sweat_smile: 

I dont recommend it. I would rather fix the rest of the app

Reply:
That would be my fault due to my ignorance... :/